### PR TITLE
operators: wire context properly to operator controllers

### PR DESCRIPTION
**Pull request by mfojtik**

### CODE REVIEW

File: pkg/operator/apiserver/controller/apiservice/apigroup.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func checkDiscoveryForAPIService(restclient rest.Interface, apiService *apiregis
 			defer wg.Done()
 			defer utilruntime.HandleCrash()
 
-			ctx, ctxCancelFn = context.WithTimeout(context.TODO(), 25*time.Second)
+			ctx, ctxCancelFn = context.WithTimeout(context.Background(), 25*time.Second)
```
Comment (sttts) [inline pkg/operator/apiserver/controller/apiservice/apigroup.go:104]: why no ctx from outside? IMO background is wrong for go routines with an wg.Wait just behind in code.

File: pkg/operator/events/eventstesting/recorder_wrapper.go [hunk 2]
```diff
@@ -11,6 +12,10 @@ type EventRecorder struct {
 	testingEventRecorder *TestingEventRecorder
 }
 
+func (e *EventRecorder) WithContext(ctx context.Context) events.Recorder {
+	return e
+}
+
```
Comment (sttts) [inline pkg/operator/events/eventstesting/recorder_wrapper.go:18]: what is this?

File: pkg/operator/staticpod/installerpod/cmd.go [hunk 3]
```diff
@@ -9,6 +9,8 @@ import (
 	"strings"
 	"time"
 
+	"k8s.io/apiserver/pkg/server"
+
```
Comment (sttts) [inline pkg/operator/staticpod/installerpod/cmd.go:13]: order

File: pkg/operator/apiserver/controller/apiservice/apigroup.go (hunk 1 above)
Comment (mfojtik) [reply pkg/operator/apiserver/controller/apiservice/apigroup.go]: fixed, wired context from the outside

File: pkg/operator/events/eventstesting/recorder_wrapper.go (hunk 2 above)
Comment (mfojtik) [reply pkg/operator/events/eventstesting/recorder_wrapper.go:18]: this is a "eventtesting" wrapper, I need to to meet the interface, but in testing we never use the context, so no need to store it.

File: pkg/operator/staticpod/installerpod/cmd.go (hunk 3 above)
Comment (mfojtik) [reply pkg/operator/staticpod/installerpod/cmd.go]: fixed.

**Pull request merged**